modules/solid_mechanics/src/vectorpostprocessors/AverageSectionValueSampler.C

```diff
@@ -151,13 +178,24 @@ AverageSectionValueSampler::finalize()
     for (const auto j : make_range(_variables.size()))
       (*_output_vector[j])[li] /= _number_of_nodes[li];
 
+  const auto pos_idx = _variables.size();
+  (*_output_vector[pos_idx]) = _positions;
+  std::vector<Real> num_nodes_real(_number_of_nodes.begin(), _number_of_nodes.end());
+  (*_output_vector[pos_idx + 1]) = num_nodes_real;
+
   if (_require_equal_node_counts)
   {
     for (const auto li : index_range(_number_of_nodes))
       if (_number_of_nodes[li] != _number_of_nodes[0])
+      {
+        std::ostringstream pos_out;
+        for (const auto li2 : index_range(_number_of_nodes))
+          pos_out << _positions[li2] << " " << _number_of_nodes[li2] << "\n";
         mooseError("Node counts do not match for all axial positions. If this behavior "
                    "is desired to accommodate nonuniform meshes, set "
-                   "'require_equal_node_counts=false'");
+                   "'require_equal_node_counts=false'\nPosition Count\n\n" +
```

Comment:
```suggestion
                   "'require_equal_node_counts=false'\nSection node counts:\n\n" +
```

Reply:
The intent here was to have "Position" and "Count" be headers for the columns of output. I changed this to be a little more clear by having it output something like:
```
     Axial      Node
  Position     Count
         0        48
         2        48
         4        48
         6        48
         8        48
        10        48